### ISSUE TITLE
`Layout/SpaceAroundOperators` behavior change since 1.54.2

### ISSUE THREAD

**Issue by yahonda** [rubocop/rubocop#12190]
## Expected behavior

It should not report any offenses as RuboCop 1.54.1 does not.
```ruby
$ rubocop -V ; rubocop foo.rb --only Layout/SpaceAroundOperators
1.54.1 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
Inspecting 1 file
.

1 file inspected, no offenses detected
$
```

## Actual behavior
It reports `Layout/SpaceAroundOperators` offense.
 
```ruby
$ rubocop foo.rb --debug --only Layout/SpaceAroundOperators
For /home/yahonda: Default configuration from /home/yahonda/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/rubocop-1.54.2/config/default.yml
Use parallel by default.
Skipping parallel inspection: only a single file needs inspection
Inspecting 1 file
Scanning /home/yahonda/foo.rb
Loading cache from /home/yahonda/.cache/rubocop_cache/a81a0ddcfa898d02fb3a6d6660d3c9a43297e217/ce8ad1600e033818c822a384669575820c6629a8/251d87e49a5c7f0a40450a45e87fb581b73ab5cb
C

Offenses:

foo.rb:5:19: C: [Correctable] Layout/SpaceAroundOperators: Operator && should be surrounded by a single space.
  return if up?   &&  migrated.include?(migration.version.to_i)
                  ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
Finished in 0.07122677500592545 seconds
$
```

## Steps to reproduce the problem

1. Create a `foo.rb`
```ruby
# frozen_string_literal: true

def execute_migration_in_transaction(migration)
  return if down? && !migrated.include?(migration.version.to_i)
  return if up?   &&  migrated.include?(migration.version.to_i)
end
```
2. Install RuboCop 1.54.2
```
gem instal rubocop -v 1.54.2
```
3. Run `rubocop foo.rb --only Layout/SpaceAroundOperators

## RuboCop version

```
$ rubocop -V
1.54.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
```

**Comment by yahonda** [comment]
This report is originally found at Rails repository by using the newer version of RuboCop. 
Here is the steps to reproduce against the Rails.

- Steps to reproduce
```ruby
git clone https://github.com/rails/rails
cd rails
rm Gemfile.lock # It has `rubocop (1.48.1)`
bundle install # to install the latest version of RuboCop
bundle exec rubocop activerecord/lib/active_record/migration.rb
```

- Actual behavior
```ruby
$ bundle exec rubocop activerecord/lib/active_record/migration.rb
Inspecting 1 file
C

Offenses:

activerecord/lib/active_record/migration.rb:1520:25: C: [Correctable] Layout/SpaceAroundOperators: Operator && should be surrounded by a single space.
        return if up?   &&  migrated.include?(migration.version.to_i)
                        ^^

1 file inspected, 1 offense detected, 1 offense autocorrectable
$ bundle exec rubocop -V
1.56.2 (using Parser 3.2.2.3, rubocop-ast 1.29.0, running on ruby 3.2.2) [x86_64-linux]
  - rubocop-md 1.2.0
  - rubocop-minitest 0.31.1
  - rubocop-packaging 0.5.2
  - rubocop-performance 1.19.0
  - rubocop-rails 2.20.2
$
```
